src/sentry/web/frontend/auth_login.py

```diff
@@ -67,11 +69,168 @@ def run_callbacks(self, request: Request):
 class AuthLoginView(BaseView):
     auth_required = False
 
-    def get_auth_provider(self, organization_slug):
-        try:
-            organization = Organization.objects.get(
-                slug=organization_slug, status=OrganizationStatus.ACTIVE
+    @never_cache
+    def handle(self, request: Request, *args, **kwargs) -> HttpResponse:
+        """
+        Hooks in to the django view dispatch which delegates request to GET/POST/PUT/DELETE.
+        Base view overwrites dispatch to include functionality for csrf, superuser, customer domains, etc.
+        """
+        return super().handle(request=request, *args, **kwargs)
+
+    def get(self, request: Request, **kwargs) -> HttpResponse:
+        next_uri = self.get_next_uri(request=request)
+        if request.user.is_authenticated:
+            return self.redirect_authenticated_user(request=request, next_uri=next_uri)
+
+        request.session.set_test_cookie()
+
+        # Initiate login clears the pending session states so you don't end up at an odd location
+        initiate_login(request=request, next_url=next_uri)
+
+        # Single org mode -- send them to the org-specific handler
+        if settings.SENTRY_SINGLE_ORGANIZATION:
+            return self.redirect_to_org_specific_login()
+
+        session_expired = "session_expired" in request.COOKIES
+        if session_expired:
+            messages.add_message(
+                request=request, level=messages.WARNING, message=WARN_SESSION_EXPIRED
+            )
+
+        default_context = self.get_default_context(request=request)
```

Comment:
`default_context = self.get_default_context(request=request)` can be moved into the functions `self.get_registration_page()` and `self.get_login_page()`.